docs/DeveloperGuide.md

```diff
@@ -336,19 +336,185 @@ the `execute("find te/alex_1")` API call.
+and `UserProfileWindow.css`. The `.fxml` file is responsible for the layout of the
+various components in this window, and the `.css` file adds a style and enhances the
+overall UI.
+
+The `UserProfileWindow` class extends `UiPart<Stage>`.
+
+This window can only be viewed when the app has successfully started up and 
+has valid User Credentials.
+
+This window is initialized when the `MainWindow` is initialized. It is
+initialized in the `MainWindow` constructor. This window, to be seen, has to be 
+triggered as an event by the user. The `MainWindow` class has a method 
+`handleUserProfileWindow()`, which is responsible for displaying this window.
+
+The `handleUserProfileWindow()` when called, calls the method `isShowing()` via 
+the object initialized earlier. If it is not showing, the `show()` method of the
+`UserProfileWindow` class is called upon. Else, if it is already showing, the
+`focus()` method is called upon. Along with that, in case the window had been
+minimized by the User, `UserProfileWindow#getRoot()#toFront()` is called to
+bring the window to the maximized state.
+
+The `UserProfileWindow#show()`, first calls the `UserProfileWindow#initializeFields()`,
+to initialize all the fields with the latest User Profile Credentials. This would, in turn
+also call `UserProfileWindow#setFields()`, to set them up in the UI. After all the setting
+up is done, the User Profile Window is shown.
+
+The `UserProfileWindow#focus()`, calls the `getRoot()` method to obtain the root object.
+On that root object, `requestFocus()` is called upon to request focus.
+
+![UserProfileWindowSequenceDiagram](images/UserProfileWindowSequenceDiagram.png)
+
+### Help Window
+
+#### Implementation
+
+The class `HelpWindow` is responsible for displaying the Help
+Window. It is shown when the user either uses the keyboard shortcut,
+`F1`, or clicks on `Help` located on the top left in
+the Menu Bar. It is facilitated by `HelpWindow.fxml`
+and `HelpWindow.css`. The `.fxml` file is responsible for the layout of the
+various components in this window, and the `.css` file adds a style and enhances the
+overall UI.
+
+The `HelpWindow` class extends `UiPart<Stage>`.
+
+This window can only be viewed when the app has successfully started up and 
+has valid User Credentials.
+
+This window is initialized when the `MainWindow` is initialized. It is
+initialized in the `MainWindow` constructor. This window, to be seen, has to be
+triggered as an event by the user. The `MainWindow` class has a method
+`handleHelpWindow()`, which is responsible for displaying this window.
+
+On initializing the `HelpWindow` class, `HelpWindow#setUpCommandDetails()` and
+`HelpWindow#setUpHelpTableView()` are called.
+
+`setUpCommandDetails()` creates multiple objects of `CommandDetails`, all of them
+representing a unique command that the app supports. Those are then added to an
+`ObservableList`, which is linked to the `TableView` in the UI.
+
```

Comment:
I believe the `ObservableList` mentioned in this paragraph should be `ObservableList<CommandDetails>` instead. 😄